.github/workflows/prover.yml

```diff
@@ -0,0 +1,55 @@
+name: Run prover guest code checks
+
+on:
+  pull_request:
+  merge_group:
+  push:
+    branches: [main]
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Set up Rust
+        uses: dtolnay/rust-toolchain@nightly
+        with:
+          toolchain: nightly-2024-07-27
+      
+      - name: Use Cargo cache
+        uses: Swatinem/rust-cache@v2
+        with:
+          cache-on-failure: true
+      
+      # Installs mold (modern ld), a drop-in replacement for lld.
+      # Under the hood, the following action symlinks mold binary onto lld,
+      # so everything is linked faster (hopefully).
+      - name: Install mold linker
+        uses: rui314/setup-mold@v1
+
+      - name: Set up Clang
+        uses: egor-tensin/setup-clang@v1
+        with:
+          version: latest
+          platform: x64
+
+      - name: Install SP1 Toolchain
+        run: |
+          curl -fsSL --proto '=https' --tlsv1.2 https://sp1.succinct.xyz | bash && ~/.sp1/bin/sp1up
+          export PATH=~/.sp1/bin:$PATH
+          cargo prove --version
+
+      - name: Checkout code
+        uses: actions/checkout@v4
+
+      - name: Build prover guest code
+        run: cargo build --profile prover-ci -F "prover"
+        working-directory: provers/sp1
+        env:
+          RUSTFLAGS: "-C link-arg=-fuse-ld=lld"
+  
+      - name: Build prover client
+        run: cargo build --profile prover-ci -F "prover"
+        working-directory: provers/sp1
+        env:
+          RUSTFLAGS: "-C link-arg=-fuse-ld=lld"
```

Comment:
What is the difference between `Build prover guest code` and `Build prover client`?
![image](https://github.com/user-attachments/assets/1683d47c-6262-4d3c-8ca6-2bb1896d7587)

Reply:
renamed.

Reply:
No, that's not it.
It is the SAME command with the SAME env vars and the SAME working dir.

Reply:
ah, right. Thanks for noticing! Fixed